### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by enniin**
First working solution, not finished. Currently running with interactive script, should perhaps be changed.

### CODE REVIEW

File: code/cdw.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    # Flatten in all but last dimension (MRI frames)
+    n_frames = voxels.shape[-1]
+    voxels = np.vstack([ voxels[:,:,:,i].flatten() for i in range(n_frames) ])
+    
+    # Calculate dissimilarity (1 - r) for each frame pair
+    rdm_bold = np.ones((n_frames,n_frames))
+    for i in range(n_frames):
+        for j in range(n_frames):
+            (r,p) = stats.pearsonr(voxels[i,:], voxels[j,:]) # mri frames are rows
+            rdm_bold[i,j] = 1 - r
```
Comment (wmvanvliet) [inline code/cdw.py]:
```suggestion
    from scipy.spatial.distance import pdist, squareform
    rdm_bold = squareform(pdist(voxels, metric='correlation'))
```

**Comment by wmvanvliet** [comment]
Great start! I was able to easily run this on my own machine without any problems, which is a major plus for an analysis pipeline.
I've added a hint on how to speed things up considerably.

The code is already very readable, so you are in a great spot to start thinking about the very best design possible. For example, I would think about the division of labour between your functions. Ideally, a function has a single task and will do that task completely, with nothing "leaking out". Meaning if a function mangles conditions, no further condition mangling should be needed either before or after the function. This is hard to achieve! But then again, we are not here to do easy things in this course ;) I'm always happy if I can have a spot somewhere in my pipeline, could be the `run_cwd_pipeline.py` file, where there is written, in code, a series of steps that are done on the data like you would write it in a methods section of a paper. Each step is a single function call that performs that step completely, taking in only obvious parameters and returning something very obvious.

**Comment by wmvanvliet** [comment]
fig1

The result I got by executing your pipeline. Pretty close already!

File: code/cdw.py (hunk 1 above)
Comment (enniin) [reply code/cdw.py]: Ahh I already made a bit if a speed-up for this (true, this version was _incredibly_ slow), but apparently forgot to push it! Anyway, I'll check these functions as well, this is way more concise.

**Comment by enniin** [comment]
Solved spherical searchlight (I think?), but still getting slightly different figures than in the assignment. Maybe a less-than/less-than-or-equal issue? Also, my kernel size is 33, not 27 – prob for the same reason.

Also reorganised function structure: the workflow is now more visible in the runner-script, which does make sense. Some hierarchically dependent steps needed to be kept "hidden" within the function script, though.

**Comment by wmvanvliet** [comment]
Good stuff! I think indeed that in the image in the assignment I did `dist < rad` and not `dist <= rad`.

**Comment by enniin** [comment]
Changed the definition of searchlight to dist < rad. Actually I think now the searchlight becomes square (as 27 = 3x3x3), which is exactly what I had in the first version! :D So the result is also identical to that.

Finished and ready for review.

**Pull request closed without merging**